src/omv/imlib/bayer.c

```diff
@@ -94,860 +94,388 @@ void imlib_debayer_line_to_binary(int x_start, int x_end, int y_row, uint32_t *d
                 row_grgr_2 = *((uint32_t *) (rowptr_grgr_2 + x - 1));
             }
 
-            #if defined(MCU_SERIES_F4) || defined(MCU_SERIES_F7) || defined(MCU_SERIES_H7)
-            int row_01 = __UHADD8(row_grgr_0, row_grgr_2);
-            int row_1g = __UHADD8(row_bgbg_1, __PKHTB(row_bgbg_1, row_bgbg_1, 16));
+            int r_pixels_0, g_pixels_0, b_pixels_0;
 
-            int r_pixels_0 = __UXTB16(__UHADD8(row_01, __PKHTB(row_01, row_01, 16)));
-            int g_pixels_0 = __UXTB16(__UHADD8(row_1g, __PKHTB(row_1g, row_01, 8)));
-            int b_pixels_0 = __UXTB16_RORn(__UHADD8(row_bgbg_1, __PKHBT(row_bgbg_1, row_bgbg_1, 16)), 8);
-            #else
+            switch (src->bpp) {
+                case IMAGE_BPP_BAYER_BGGR: {
+                    #if defined(MCU_SERIES_F4) || defined(MCU_SERIES_F7) || defined(MCU_SERIES_H7)
```

Comment:
```suggestion
#if defined(ARM_MATH_DSP)
```

You should double check this, but the right macro to test for before using these functions is `#if (__ARM_FEATURE_DSP == 1) see cmsis_gcc.h. The MCU_SERIES_xx macros are for specific to stm32, ex: if NXP is supported this won't work, especially in common/shared code.

Reply:
You are right. Should replace this in other places too.